aiohttp/pytest_plugin.py

```diff
@@ -227,19 +228,22 @@ def test_client(loop):
     clients = []
 
     @asyncio.coroutine
-    def go(__param, *args, **kwargs):
-        if isinstance(__param, Application):
-            assert not args, "args should be empty"
-            client = TestClient(__param, loop=loop, **kwargs)
-        elif isinstance(__param, TestServer):
-            assert not args, "args should be empty"
-            client = TestClient(__param, loop=loop, **kwargs)
-        elif isinstance(__param, RawTestServer):
+    def go(__param, *args, server_kwargs={}, **kwargs):
+
+        if isinstance(__param, collections.Callable) and \
+                not isinstance(__param, (Application, BaseTestServer)):
+            __param = __param(loop, *args, **kwargs)
+            kwargs = {}
+        else:
             assert not args, "args should be empty"
+
+        if isinstance(__param, Application):
+            server = TestServer(__param, loop=loop, **server_kwargs)
+            client = TestClient(server, loop=loop, **kwargs)
+        elif isinstance(__param, BaseTestServer):
```

Comment:
I dropped here the duplicated test on TestServer and RawTestServer. The argument to TestClient must now be a BaseTestServer instance which includes both. Maybe someone would want to make a TestServer not based on TestServer but on BaseTestServer